kratos/sources/kratos_application.cpp

```diff
@@ -229,90 +204,36 @@ void KratosApplication::RegisterKratosCore() {
-
-    Tetrahedra3D10<NodeType > Tetrahedra3D10Prototype( GeometryType::PointsArrayType(10));
-    Serializer::Register("Tetrahedra3D10", Tetrahedra3D10Prototype);
-
-    //Prisms:
-    Prism3D6<NodeType > Prism3D6Prototype( GeometryType::PointsArrayType(6));
-    Serializer::Register("Prism3D6", Prism3D6Prototype);
-
-    Prism3D15<NodeType > Prism3D15Prototype( GeometryType::PointsArrayType(15));
-    Serializer::Register("Prism3D15", Prism3D15Prototype);
-
-    //Hexahedra:
-    Hexahedra3D8<NodeType > Hexahedra3D8Prototype( GeometryType::PointsArrayType(8));
-    Serializer::Register("Hexahedra3D8", Hexahedra3D8Prototype);
-
-    Hexahedra3D20<NodeType > Hexahedra3D20Prototype( GeometryType::PointsArrayType(20));
-    Serializer::Register("Hexahedra3D20", Hexahedra3D20Prototype);
-
-    Hexahedra3D27<NodeType > Hexahedra3D27Prototype( GeometryType::PointsArrayType(27));
-    Serializer::Register("Hexahedra3D27", Hexahedra3D27Prototype);
-
-    //QuadraturePointGeometry:
+    // Point register:
+    Serializer::Register("Point", mPointPrototype);
+    
+    // Register + KratosComponents
+    KRATOS_REGISTER_GEOMETRY("Point2D", mPoint2DPrototype);
+    KRATOS_REGISTER_GEOMETRY("Point3D", mPoint3DPrototype);
+    KRATOS_REGISTER_GEOMETRY("Sphere3D1", mSphere3D1Prototype);
+    KRATOS_REGISTER_GEOMETRY("Line2D2", mLine2D2Prototype);
+    KRATOS_REGISTER_GEOMETRY("Line2D3", mLine2D3Prototype);
+    KRATOS_REGISTER_GEOMETRY("Line3D2", mLine3D2Prototype);
+    KRATOS_REGISTER_GEOMETRY("Line3D3", mLine3D3Prototype);
+    KRATOS_REGISTER_GEOMETRY("Triangle2D3", mTriangle2D3Prototype);
+    KRATOS_REGISTER_GEOMETRY("Triangle2D6", mTriangle2D6Prototype);
+    KRATOS_REGISTER_GEOMETRY("Triangle3D3", mTriangle3D3Prototype);
+    KRATOS_REGISTER_GEOMETRY("Triangle3D6", mTriangle3D6Prototype);
+    KRATOS_REGISTER_GEOMETRY("Quadrilateral2D4", mQuadrilateral2D4Prototype);
+    KRATOS_REGISTER_GEOMETRY("Quadrilateral2D8", mQuadrilateral2D8Prototype);
+    KRATOS_REGISTER_GEOMETRY("Quadrilateral2D9", mQuadrilateral2D9Prototype);
+    KRATOS_REGISTER_GEOMETRY("Quadrilateral3D4", mQuadrilateral3D4Prototype);
+    KRATOS_REGISTER_GEOMETRY("Quadrilateral3D8", mQuadrilateral3D8Prototype);
+    KRATOS_REGISTER_GEOMETRY("Quadrilateral3D9", mQuadrilateral3D9Prototype);
+    KRATOS_REGISTER_GEOMETRY("Tetrahedra3D4", mTetrahedra3D4Prototype);
+    KRATOS_REGISTER_GEOMETRY("Tetrahedra3D10", mTetrahedra3D10Prototype);
+    KRATOS_REGISTER_GEOMETRY("Prism3D6", mPrism3D6Prototype);
+    KRATOS_REGISTER_GEOMETRY("Prism3D15", mPrism3D15Prototype);
+    KRATOS_REGISTER_GEOMETRY("Hexahedra3D8", mHexahedra3D8Prototype);
+    KRATOS_REGISTER_GEOMETRY("Hexahedra3D20", mHexahedra3D20Prototype);
+    KRATOS_REGISTER_GEOMETRY("Hexahedra3D27", mHexahedra3D27Prototype);
+    
+    //QuadraturePointGeometry register:
     QuadraturePointGeometry<NodeType, 3> QuadraturePointGeometryVolume3dPrototype(GeometryType::PointsArrayType(1));
```

Comment:
Btw. is there an issue with the QuadraturePointGeometries?

Reply:
Yes, give a memory error with the register. i will need more time to fix it...

Reply:
Ok, I will give it a try, too

Reply:
I added it, it worked with raw pointers